src/lyra/engine/runner.py

```diff
@@ -84,17 +84,18 @@ def variables(self) -> List[VariableIdentifier]:
                     worklist.put(node)
         return variables
 
-    def main(self, path):
+    def main(self, path, do_render=True):
         self.path = path
         with open(self.path, 'r') as source:
             self.source = source.read()
             self.tree = ast.parse(self.source)
             self.cfg = ast_to_cfg(self.tree)
-        self.run()
+        return self.run(do_render)
 
-    def run(self) -> AnalysisResult:
+    def run(self, do_render=True) -> AnalysisResult:
         result = self.interpreter().analyze(self.state())
-        self.render(result)
+        if do_render:
+            self.render(result)
```

Comment:
Do not change things here. You can override things in your 'AssumptionAnalysis(Runner)'. Specifically, you can override 'render' using 'view=False' in the last line.